drafts/Asynchronous-programming-made-resilient-with-MPFT.adoc

```diff
@@ -0,0 +1,230 @@
+
+[source,java]
+----
+@Asynchronous
+public CompletionStage<String> serviceA() { 
+
+	return CompletableFuture.completedFuture("serviceA");
+
+}
+----
+
+What does this do when we call the method?
+
+Well, usually when you call a method, you call it, and then it returns a result. However, when we use the `@Asynchronous` annotation, Fault Tolerance gets in between. When you call the method, Fault Tolerance first schedules your method to run later on a different thread, then it returns you a `CompletionStage` or `Future`. This isn't the `CompletionStage` or `Future` that's returned from your method, because your method hasn't run yet, it's an object created entirely by Fault Tolerance. 
+
+image::/img/blog/FT-basic-asynchronous-execution.png[Fault Tolerance asynchronous execution flow]
+
+Sometime later, your method runs and returns a `CompletionStage` or `Future`. Now, Fault Tolerance will go and complete the `CompletionStage` or `Future` object which was returned to the caller earlier, with the value that was returned from your method. You might be wondering *_"What if the returned result is an exception? What can I do?"_*.  Read along.
+
+== How you can use `@Asynchronous`
+So `@Asynchronous` sounds great, right? But how are we going to use it to make our asynchronous programs resilient? Let's go through 2 use-cases:
+
+1. Applying Fault Tolerance to an API which is already asynchronous
+2. Using `@Asynchronous` annotation to do two things in parallel with resilience
+
+=== 1. Applying to An Already Asynchronous API
+Another use is to apply Fault Tolerance to an API which is already asynchronous. We can do this with any API which does something Asynchronously and returns a `CompletionStage`.
+
+In this example, we'll use the JAX-RS client API for calling remote REST services, introduced in JAX-RS 2.1. We can build up a request to fetch a String from a given URL with a GET request and the return type is a `CompletionStage` of String:
+
+==== JAX-RS client API `CompletionStage` example:
+
+[source,java]
+----
+private Client client = ClientBuilder.newClient();
+
+public CompletionStage<String> clientDemo() {
+
+	CompletionStage<String> response = client.target("http://example.com/resource")
+					.request(MediaType.TEXT_PLAIN)
+					.rx()
+					.get(String.class);
+
+	return response;
+
+}
+----
+==== Fetching the following resource:
+[source,java]
+----
+@GET
+@Path("/resource")
+public CompletionStage<String> fetchExample() { 
+
+	return CompletableFuture.completedFuture("responseText");
+
+}
+----
+
+If we call the `clientDemo()` method, without any annotations, it works as we expect. We call it, receive a `CompletionStage<String>` (named `response` in the example), and then we can add an action to take when the `CompletionStage` completes.
+
+[source,java]
+----
+response.thenAccept(System.out::println);
+--> responseText
+----
+
+However, what happens if we want to use the `@Retry` annotation with our method? The answer is nothing. 
```

Comment:
The answer is nothing. => The answer is that the method will not be retried. Here is why.

Reply:
I agree, this is a little more clear.